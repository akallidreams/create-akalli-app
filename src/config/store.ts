import { AkState } from "@akalli/state";
import { authSlice } from "../features/auth/helpers/redux";

export const { clearStore, store, useAppDispatch, persistor } =
  AkState.reduxInit({
    auth: authSlice.reducer,
  });
