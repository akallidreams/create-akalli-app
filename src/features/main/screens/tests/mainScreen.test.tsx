import { Main } from "../";
import { render, screen } from "@testing-library/react-native/pure";
import { store } from "@config/store";
import { Provider } from "react-redux";
import "@testing-library/jest-native/extend-expect";
import React, { ReactNode } from "react";

jest.mock("redux-persist", () => {
  const real = jest.requireActual("redux-persist");
  return {
    ...real,
    persistReducer: jest
      .fn()
      .mockImplementation((config, reducers) => reducers),
  };
});

const ReduxWrapper = ({ children }: { children: ReactNode }) => (
  <Provider store={store}>{children}</Provider>
);

describe("should test Main Screen", () => {
  beforeEach(() => {
    render(
      <ReduxWrapper>
        <Main />
      </ReduxWrapper>
    );
  }),
    it("renders without crashing", () => {
      expect(screen).toBeTruthy();
    });
  it("should render the title value", () => {
    const { queryByTestId } = render(
      <ReduxWrapper>
        <Main />
      </ReduxWrapper>
    );
    expect(queryByTestId("dti-title")).toHaveTextContent("Logged in");
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
