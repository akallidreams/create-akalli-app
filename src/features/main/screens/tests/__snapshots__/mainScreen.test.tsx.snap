// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test Main Screen should render the title value 1`] = `
<View
  height="100%"
  style={
    Array [
      Object {
        "height": "100%",
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    testID="dti-title"
  >
    Logged in
  </Text>
</View>
`;
